src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js

```diff
@@ -49,6 +53,7 @@ class BaseReportActionContextMenu extends React.Component {
                             draftMessage: this.props.draftMessage,
                             selection: this.props.selection,
                         })}
+                        description={ContextMenuUtils.getPopoverDescription(this.props.type, this.props.selection, this.props.isSmallScreenWidth)}
```

Comment:
Did you notice how most of the props are coming from `contextAction`? This change is breaking the existing pattern. This is a side-effect. 


Let's discuss the alternative first before making the change. Consider that the only way to present data here is to get it from `ContextMenuActions`. Now, what can be done to solve this problem?

Look at this https://github.com/Expensify/App/blob/31e49ec54110b8684de7ccc3f7c1d1811d7dd760/src/pages/home/report/ContextMenu/ContextMenuActions.js#L60 . How are we picking between actions? 

Can we show a new item here dynamically? 

What if we have to show different descriptions for different menu items. Will your solution work in an ad-hoc way?

Reply:
> Consider that the only way to present data here is to get it from `ContextMenuActions`. Now, what can be done to solve this problem?
> Can we show a new item here dynamically?
> What if we have to show different descriptions for different menu items. Will your solution work in an ad-hoc way?

We could
1. Add `getDescription` method [here](https://github.com/Expensify/App/blob/31e49ec54110b8684de7ccc3f7c1d1811d7dd760/src/pages/home/report/ContextMenu/ContextMenuActions.js#L64)
```javascript
  {
      textTranslateKey: 'reportActionContextMenu.copyURLToClipboard',
      icon: Expensicons.Clipboard,
      successTextTranslateKey: 'reportActionContextMenu.copied',
      successIcon: Expensicons.Checkmark,
      shouldShow: type => type === CONTEXT_MENU_TYPES.LINK,
      onPress: (closePopover, {selection}) => {
          Clipboard.setString(selection);
          hideContextMenu(true, ReportActionComposeFocusManager.focus);
      },
      getDescription: ({selection}) => selection,
  },
```
2. Call it from `BaseReportActionContextMenu` [here](https://github.com/Expensify/App/blob/31e49ec54110b8684de7ccc3f7c1d1811d7dd760/src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js#L52)
```javascript
description={contextAction.getDescription && contextAction.getDescription({selection: this.props.selection})}
```
And if we will need to show different description for, say, `CONTEXT_MENU_TYPES.EMAIL`, we could add a `getDescription` method to it too
```javascript
getDescription: ({otherParam}) => otherParam,
```
Then step 2 would change to
```javascript
description={contextAction.getDescription && contextAction.getDescription({selection: this.props.selection, otherParam: 'lorem ipsum'})}
```

Would this work as an alternative?

Reply:
Sounds good to me. How will you control to hide the description on Desktop | Web?

Reply:
> Sounds good to me. How will you control to hide the description on Desktop | Web?

I would implement `shouldShowDescription` method in `src/pages/home/report/ContextMenu/ContextMenuUtils`, for native it would just return `true`, and for web it would look like this:
```javascript
function shouldShowDescription(isSmallScreenWidth) {
    return isSmallScreenWidth;
}
```
But I'm not sure where to call it
We could call it in `getDescription` method for each action type individually, like this:
```javascript
getDescription: ({selection}) => shouldShowDescription() ? selection : '',
```
But i think it makes more sense to call it for all actions in BaseReportActionContextMenu?
```javascript
description={(contextAction.getDescription && ContextMenuUtils.shouldShowDescription()) && contextAction.getDescription({selection: this.props.selection})}
```

Reply:
What if we pass the isSmallScreenWidth directly to the description callback and decide to show the value here. 
`getDescription: ({selection}) => shouldShowDescription() ? selection : '',`

Reply:
I guess so.